app/workers/metrics/record_data_counts_worker.rb

```diff
@@ -9,6 +9,11 @@ def perform
         estimate = model.estimated_count
         Rails.logger.info("db_table_size", table_info: { table_name: model.table_name, table_size: estimate })
         DatadogStatsClient.gauge("postgres.db_table_size", estimate, tags: { table_name: model.table_name })
+
+        next unless model.const_defined?(:SEARCH_CLASS)
+
+        index_size = Search::Client.count(index: model::SEARCH_CLASS::INDEX_ALIAS).dig("count")
+        DatadogStatsClient.gauge("elasticsearch.index_size", index_size, tags: { table_name: model.table_name })
```

Comment:
Can we rename this gauge `"elasticsearch.document_count"`? Index size to me means the amount of space in bytes the index itself takes up.